gregBorchers/hwWeek1.js

```diff
@@ -0,0 +1,215 @@
+// circumference is 2*PI*R
+var maxCircumference = 2 * Math.PI * ((height < width ? height : width )/2) // max circumference is based on radius inscribed inside whichever is smaller height or width
+
+// **d)**
+// Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+// area is PI*R^2 distance is sqrt( (x2-x1)^2 + (y2-y1)^2 )
+//var minRadius = Math.sqrt( Math.pow((x2-x1),2) + Math.pow((y2-y1),2)) / 2; // half the diagonal is the minimal radius of the circumscribing circle
+var minRadius = (((x2-x1) < y2-y1) ? x2-x1 : y2-y1 );
+var minArea   = Math.PI * Math.pow((minRadius),2); 
+
+
+// **e)**
+// Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+// Define four new variables describing the centermost small rectangle.
+// (_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+
+var centerX1 // left side of center rectangle
+var centerX2 // right side
+var centerY1 // lower side
+var centerY2 // upper side
+
+centerX1 = x1 + ((x2-x1)/3);
+centerX2 = centerX1 + ((x2-x1)/3);
+centerY1 = y1 + ((y2-y1)/3);
+centerY2 = centerY1 + ((y2-y1)/3);
+
+// ---
+
+// **4)** (_Difficulty: easyish_)
+
+// Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+// * Each square has integer coordinates _(R,C)_ describing its row and column.  Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+// * Each square has a unique integer number N from 0 to 63.  These numbers run sequentially left-to-right one row at a time, top to bottom.  Therefore the upper-left square has N==0 and the bottom-right has N==63.
+
+// **a)**  Given a particular R and C, find the corresponding N.  That is, write an expression for variable N containing variables R and C.
+	var row = 5
+	var col = 2
+	var n = (row * 8) + (col);
+
+
+
+// **b)**  Given N, find R.  Write an expression for R which contains N.
+var row = Math.trunc( n / 8 );
+
+
+// **c)**  Given N, find C.  Write an expression for C which contains N.
+var col = n % 8;
+
+
+// **d)**  Assume the squares are colored black and white, with the upper-left square black.
+// Write an expression to set a variable _color_ to either 'black' or 'white', describing the square identified by variables R,C, and N.
+// (_Hint: If you prefer, you may use a pair of conditional statements instead of a single expression._)
+
+//  squareColor = black if (even row AND even column) OR (odd row AND odd column)
+
+
+	if      (((Math.trunc( n / 8 ) % 2) == 0) && (( n % 8 ) % 2) == 0) // row and col are both even
+	{ 
+		squareColor = "black"
+	} 
+	else if (((Math.trunc( n / 8 ) % 2) != 0) && (( n % 8 ) % 2) != 0) // row and col are both odd
+	{
+		squareColor = "black"
+	} 
+	else
+	{
+		squareColor = "white"
+	};
```

Comment:
Good! Remember that you can re-use your variables; it's possible (and more readable) to re-use `row` instead of `Math.trunc( n / 8 )`